nncf/experimental/tensor/functions/numpy_numeric.py

```diff
@@ -170,8 +170,14 @@ def _(a: np.ndarray, source: Union[int, Tuple[int, ...]], destination: Union[int
 
 
 @register_numpy_types(numeric.mean)
-def _(a: Union[np.ndarray, np.generic], axis: Union[int, Tuple[int, ...]] = None, keepdims: bool = False) -> np.ndarray:
-    return np.array(np.mean(a, axis=axis, keepdims=keepdims))
+def _(
+    a: Union[np.ndarray, np.generic],
+    axis: Union[int, Tuple[int, ...]] = None,
+    keepdims: bool = False,
+    dtype: Optional[TensorDataType] = None,
+) -> np.ndarray:
+    dtype = DTYPE_MAP[dtype] if dtype else None
+    return np.array(np.mean(a, axis=axis, keepdims=keepdims, dtype=dtype))
```

Comment:
Most likely I missed the discussion, but what are the reasons behind this change? Why it is applicable only to the mean operation?

Reply:
The reason is that during experiments with FBC and BC it was observed that calculation of mean for batch_size > 1 at float32 precision leads to inaccurate results. 
According documentation https://numpy.org/doc/stable/reference/generated/numpy.mean.html it is recommended to switch to a higher precision and it really helped. 
However, I reverted changes to BC and FBC I keeped dtype option as it could be usefull elsewhere.